Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/PublicBlogController.java

```diff
@@ -0,0 +1,153 @@
+        model.addAttribute("project", project);
+        model.addAttribute("project_id", project.getProjectId());
+
+        return "sites/projectblog";
+    }
+
+    /**
+     * This method provides page to add project blog entry to the authorized
+     * user. Only project owners are allowed to add entry to the blog.
+     * 
+     * @param projectUnixName
+     *            The project Unix name
+     * @param projectId
+     *            The id assigned to project by framework
+     * @param model
+     *            Model instance
+     * @return view to add new project blog entry
+     * @throws QuadrigaStorageException
+     * @throws QuadrigaAccessException
+     *             This exception is thrown when user other than project admin
+     *             and owner tries to add project blog entry.
+     */
+    @AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.PROJECT, paramIndex = 1, userRole = {
+            RoleNames.ROLE_COLLABORATOR_ADMIN, RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN }) })
+    @RequestMapping(value = "sites/{projectId}/addprojectblog", method = RequestMethod.GET)
+    public String addprojectblog(@PathVariable("projectId") String projectId, Model model)
+                    throws QuadrigaStorageException, QuadrigaAccessException {
+
+        IProject project = projectManager.getProjectDetails(projectId);
+
+        if (project == null) {
+            return "forbidden";
+        }
+
+        model.addAttribute("projectBlog", new ProjectBlog());
+        model.addAttribute("project", project);
+        model.addAttribute("project_id", projectId);
+        model.addAttribute("success", 0);
+
+        return "sites/addprojectblog";
+    }
+
+    @AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.PROJECT, paramIndex = 3, userRole = {
+            RoleNames.ROLE_COLLABORATOR_ADMIN, RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN }) })
+    @RequestMapping(value = "sites/{projectId}/addprojectblog", method = RequestMethod.POST)
+    public String addprojectblogEntry(@Validated @ModelAttribute("projectBlog") ProjectBlog projectBlog,
+            BindingResult result, @PathVariable("projectId") String projectId, Model model, Principal principal)
+                    throws QuadrigaStorageException, QuadrigaAccessException {
+
+        // The ProjectBlog object contains title and description of the blog
+        // obtained from user interface
+        // Now, populating the ProjectBlog object with the following information
+        // setting author which is a user who is currently logged in
+        projectBlog.setAuthor(principal.getName());
+        // setting project id of the blog
+        projectBlog.setProjectId(projectId);
+        // setting date of creation for the blog
+        projectBlog.setCreatedDate(new Date());
+
+        ProjectBlogDTO projectBlogDTO = projectBlogDTOMapper.getProjectBlogDTO(projectBlog);
+        
+        projectBlogDao.addProjectBlogDTO(projectBlogDTO);
```

Comment:
you are missing a manager class. you controllers should talk to a manager how talks to the DAO. I would also let the manager set the creation date and maybe author, so that if there is a second way of creating blog entries, this code doesn't have to be repeated.